fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/Index.java

```diff
@@ -65,6 +65,8 @@
     @Nonnull
     private final Map<String, String> options;
     @Nonnull
+    private final Map<Integer, Map<String, String>> fieldOptions;
```

Comment:
Actually I just prefer to have the options with the field of the record metadata rather than the mapped fields in the key expression. One example is the ```LuceneDocumentFromRecordTest.map()```. In this case, I want the options to be correlated with the field ```entry``` rater than the ```k1``` or ```k2```. So the clients can simply define the options in the record metadata without considering what will the mapped filed names look like.